js/app/test/cancel_events.spec.ts

```diff
@@ -44,3 +45,26 @@ test("when using an iterative function it should be possible to cancel the funct
 	await page.waitForTimeout(1000);
 	await expect(textbox).toHaveValue("0");
 });
+
+test("when using an iterative function and the user closes the page, the python function should stop running", async ({
+	page
+}) => {
+	const start_button = await page.locator("button", {
+		hasText: /Start Iterating/
+	});
+
+	await start_button.click();
+	await page.waitForTimeout(1000);
+	await page.close();
+
+	// wait for the duration of the entire iteration
+	// check that the final value did not get written
+	// to the log file. That's our proof python stopped
+	// running
+	await new Promise((resolve) => setTimeout(resolve, 5000));
+	const data = readFileSync(
+		"../../demo/cancel_events/cancel_events_output_log.txt",
+		"utf-8"
+	);
+	expect(data).not.toContain("Current step: 8");
```

Comment:
Let's also ensure that at least one of the steps that should be logged are logged
```suggestion
	expect(data).not.toContain("Current step: 1");
	expect(data).not.toContain("Current step: 8");
```

Reply:
Reasonable!